### PR TITLE
Fix #3585, remove misleading buffer too small messages

### LINKED ISSUE #3585: Misleading OE_BUFFER_TOO_SMALL error messages
There are many benign OE_BUFFER_TOO_SMALL errors in traces from functions which return required buffer size. Such functions are typically called twice - first time to get required size; second time to retrieve data with allocated buffer. The first call unnecessarily prints OE_BUFFER_TOO_SMALL error messages that can be misleading when root causing issue from log.

Example:
```c++
    r = oe_get_crl_distribution_points(..., NULL, &buffer_size);
    if (r == OE_BUFFER_TOO_SMALL)
    {
        uint8_t* buffer = (uint8_t*)malloc(buffer_size);
        oe_get_crl_distribution_points(..., buffer, &buffer_size);
```

### PR DESCRIPTION
**Pull request by qiucwang**
Fixes #3585 

Signed-off-by: Qiucheng Wang <qiucwang@microsoft.com>

### CODE REVIEW

**Comment by yentsanglee** [comment]
Log at info level rather than error level if the caller passes in a null buffer into these functions.

File: host/crypto/openssl/key.c [hunk 1]
```diff
@@ -347,7 +347,7 @@ oe_result_t oe_public_key_write_pem(
         if (*size < mem->length)
         {
             *size = mem->length;
-            OE_RAISE(OE_BUFFER_TOO_SMALL);
+            OE_RAISE_NO_TRACE(OE_BUFFER_TOO_SMALL);
```
Comment (yentsanglee) [inline host/crypto/openssl/key.c:350]: I believe in this case you want trace.

**Review by yentsanglee — APPROVED** [review]

File: host/crypto/openssl/key.c (hunk 1 above)
Comment (qiucwang) [reply host/crypto/openssl/key.c:350]: Thanks for catching that. Miss this one

**Comment by qiucwang** [comment]
bors try

**Comment by qiucwang** [comment]
bors try

**Comment by qiucwang** [comment]
bors retry

**Review by ryanhsu19 — APPROVED** [review]

**Comment by yentsanglee** [comment]
bors retry

**Pull request merged**